source/fundamentals/data-formats/document-data-format-extended-json.txt

```diff
@@ -0,0 +1,250 @@
+
+Extended JSON Examples
+~~~~~~~~~~~~~~~~~~~~~~
+
+The following examples show a document containing an ObjectId, date, and long
+number field represented in each Extended JSON format. Click the tab that
+corresponds to the format of the example you want to see:
+
+.. tabs::
+
+   .. tab:: Extended
+      :tabid: extended-format
+
+      .. code-block:: json
+
+         {
+           "_id": { "$oid": "573a1391f29313caabcd9637" },
+           "createdAt": { "$date": { "$numberLong": "1601499609" }},
+           "numViews": { "$numberLong": "36520312" }
+         }
+
+   .. tab:: Relaxed
+      :tabid: relaxed-format
+
+      .. code-block:: json
+
+         {
+           "_id": { "$oid": "573a1391f29313caabcd9637" },
+           "createdAt": { "$date": "2020-09-30T18:22:51.648Z" },
+           "numViews": 36520312
+         }
+
+   .. tab:: Shell
+      :tabid: shell-format
+
+      .. code-block:: json
+
+         {
+           "_id:": ObjectId("573a1391f29313caabcd9637"),
+           "createdAt": ISODate("2020-09-30T18:22:51.648Z"),
+           "numViews": NumberLong("36520312")
+         }
+
+   .. tab:: Strict
+      :tabid: strict-format
+
+      .. code-block:: json
+
+         {
+           "_id": { "$oid": "573a1391f29313caabcd9637" },
+           "createdAt": { "$date": "2020-09-30T18:22:51.648Z" },
+           "numViews": { "$numberLong": "36520312" }
+         }
+
+
+Read Extended JSON
+------------------
+
+You can read an Extended JSON string into Java objects with the BSON library
+by using an instance of the :java-docs:`JsonReader </apidocs/bson/org/bson/json/JsonReader.html>`
+class. This class contains methods to sequentially parse the fields and values
+in the Extended JSON object, and returns them as Java objects. The
+``JsonReader`` class automatically handles any of the Extended JSON formats.
+
+The following code example shows how you can use ``JsonReader`` to convert
+an Extended JSON string into Java objects:
+
+.. code-block:: java
+
+   String ejsonStr = "{ \"_id\": { \"$oid\": \"507f1f77bcf86cd799439011\"}," +
+                     "\"myNumber\": {\"$numberLong\": \"4794261\" }}}";
+
+   JsonReader jsonReader = new JsonReader(ejsonStr);
+
+   jsonReader.readStartDocument();
+
+   jsonReader.readName("_id");
+   ObjectId id = jsonReader.readObjectId();
+   jsonReader.readName("myNumber");
+   Long myNumber = jsonReader.readInt64();
+
+   jsonReader.readEndDocument();
+
+   System.out.println(id + " is type: " + id.getClass().getName());
+   System.out.println(myNumber + " is type: " + myNumber.getClass().getName());
+
+   jsonReader.close();
+
+The output of this code example resembles the following:
+
+.. code-block:: none
+   :copyable: False
+
+   507f1f77bcf86cd799439011 is type: org.bson.types.ObjectId
+   4794261 is type: java.lang.Long
+
+To construct an instance of a ``Document`` from Extended JSON, you can call
```

Comment:
Document.parse() is going to be the way that almost everyone will parse JSON.  Should we lead with that and leave JsonReader usage to the end?

Same goes for Document.toJson() vs. JsonWriter

Reply:
Ah, I thought it could be important to explain how the reader or writer works first to provide them with the context of how it works. I don't have a strong preference here; I'll update to present the Document methods first and see how it looks.

Reply:
Addressed in f154afda39c6d5d83d850a77618fb0a8673b4eb5 and b4e42b9e7d6162577786543786ae09e64207102b